Test_Automation/README

```diff
@@ -0,0 +1,75 @@
+# * Copyright (c) 2020-2021, Happiest Minds Technologies Limited Intellectual Property. All rights reserved.
+# *
+# * SPDX-License-Identifier: LGPL-2.1-only
+
+# This section is divided in to 3 parts:
+# (a) Automation Framework structure
+# (b) Installation procedure
+# (c) Test script execution procedure
+
+(a) Automation framework structure:
+   README   :- This file
+   script   :- Contains main test scripts. Test scripts can be added in this directory
+   keyword  :- Contains keywords which are required for test scripts. Keywords can be re-used for scripts and new keywords can be added to this directory
+   library  :- Contains python supporting library functions which are required for different command execution and it can be re-used for different test scripts
+   variable :- Contains variables required for test scripts
+   testdata :- Contains test data required for the test script execution. User has to modify the respective test data file under this directory based on the topology created, before execution of test script
```

Comment:
Do you think this directory structure will scale for multiple test suites? So for example if I add tests for DHCP, SNMP, OSPF, RIP and so on will it still be easy to find the variables/testdata/keywords that apply to a script?

Would it be easier to have your scripts directory include feature sub directories - for example,

scripts
+----IPSEC_VPN
       + testscript.robot
       + keyword
       + testdata
       + variable
+----MPLS_VPN
+----DHCP
+---- etc

And for each of these directories to contain robot test/variables/testdata/keywords which are specific to that script

Reply:
Hi Trumpton,
Yes. the framework is scalable. I want to keep the directories as is because keywords also can be re-used in different test scripts of the use cases scripts that are being developed. We can keep the test scripts names under the script directory as the current focus of designing this framework is to cover multiple use cases.
But, definitely, the idea of keeping different directories for each feature will be good so that the individual feature test cases can be added under that directory. That will be like CSIT framework and it will be good to test each features as regression test suites. It can be planed once we have a good number of test cases for each feature.